code/modules/ruins/lavaland_anomalies.dm

```diff
@@ -0,0 +1,80 @@
+//Black Box
+
+/obj/machinery/smartfridge/black_box
+	name = "black box"
+	desc = "A completely indestructible chunk of crystal, rumoured to predate the start of this universe. It looks like you could store things inside it."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_on = "blackbox"
+	icon_off = "blackbox"
+	luminosity = 8
+	max_n_of_items = 10
+	pixel_y = -4
+	use_power = 0
+	var/duplicate = FALSE
+	var/memory_saved = FALSE
+	var/list/stored_items = list()
+
+/obj/machinery/smartfridge/black_box/accept_check(obj/item/O)
+	if(istype(O, /obj/item))
+		return 1
+	return 0
+
+/obj/machinery/smartfridge/black_box/New()
+	..()
+	for(var/obj/machinery/smartfridge/black_box/B in machines)
+		if(B != src)
+			duplicate = 1
+			qdel(src)
+	ReadMemory()
+
+/obj/machinery/smartfridge/black_box/process()
+	..()
+	if(ticker.current_state == GAME_STATE_FINISHED && !memory_saved)
+		WriteMemory()
+
+/obj/machinery/smartfridge/black_box/proc/WriteMemory()
+	var/savefile/S = new /savefile("data/npc_saves/Blackbox.sav")
+	stored_items = list()
+	for(var/obj/I in component_parts)
+		qdel(I)
+	for(var/obj/O in contents)
+		stored_items += O.type
```

Comment:
you don't even have to save the types, you can save entire objects to the savefile and load them perfectly, DM doesn't give a shit.